src/structures/BaseMessageComponent.js

```diff
@@ -43,6 +46,15 @@ class BaseMessageComponent {
    * @param {BaseMessageComponent|BaseMessageComponentOptions} [data={}] The options for this component
    */
   constructor(data) {
+    if (!emittedDeprecationNotice) {
+      process.emitWarning(
+        'BaseMessageComponent is deprecated and will be removed in a future release, please use BaseComponent instead',
```

Comment:
Matches the ones from #6994 
```suggestion
        'The BaseMessageComponent class is deprecated. Use BaseComponent instead',
```